File: backend/models/Task.js
const mongoose = require('mongoose');

const TaskSchema = new mongoose.Schema({
  projectId:      { type: mongoose.Schema.Types.ObjectId, ref: 'Project' },
  title:          { type: String, required: true },
  description:    { type: String },
  assignedTo:     { type: mongoose.Schema.Types.ObjectId, ref: 'User' },
  assignedToName: { type: String },
  status:         { type: String, default: 'pending' },
  deadline:       { type: String },
  file:           { type: String },
  comment:        { type: String }
}, { timestamps: true });

module.exports = mongoose.model('Task', TaskSchema);

File: backend/src/main/java/com/projecthub/model/Task.java
package com.projecthub.model;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Data
@Document(collection = "tasks")
public class Task {
    @Id
    private String id;
    
    private String projectId;
    private String title;
    private String description;
    private String assignedTo;
    private String assignedToName;
    private String status; // pending, in-progress, completed
    private String deadline;
    private String fileUrl;
    private String comments;
    private String createdAt;
}
